nncf/experimental/torch/fx/transformations.py

```diff
@@ -0,0 +1,398 @@
+                # For scale and zero_point values we register them as buffers in the root module.
+                # However, note that when the values are not tensors, as in the case of
+                # per_tensor quantization, they will be treated as literals.
+                # However, registering them as a node seems to cause issue with dynamo
+                # tracing where it may consider tensor overload as opposed to default.
+                # With extra check of scale and zero_point being scalar, it makes
+                # sure that the default overload can be used.
+                # TODO: maybe need more complex attr name here
+                qparam_node = create_getattr_from_value(model, graph, target_node.name + key, value_or_node)
+                quantize_op_inputs.append(qparam_node)
+            else:
+                # for qparams that are not scale/zero_point (like axis, dtype) we store
+                # them as literals in the graph.
+                quantize_op_inputs.append(value_or_node)
+    with graph.inserting_after(target_node):
+        quantized_node = graph.create_node(node_type, quantize_op, tuple(quantize_op_inputs), {})
+        # use the same qparams from quantize op
+    dq_inputs = [quantized_node] + quantize_op_inputs[1:]
+    user_dq_nodes = []
+    with graph.inserting_after(quantized_node):
+        for user in target_node.users:
+            if user is quantized_node:
+                continue
+            user_dq_nodes.append((user, graph.call_function(dequantize_op, tuple(dq_inputs), {})))
+
+    for user, dq_node in user_dq_nodes:
+        user.replace_input_with(target_node, dq_node)
+
+
+def _insert_call_module(graph: torch.fx.Graph, target_node: torch.fx.Node, module_attr_name: str):
+    """
+    Inserts module call node to the graph after the target node.
+
+    :param graph: Graph to insert module call node.
+    :param target_node: Target node, module call node is being iserted just after the target node.
+    :param module_attr_name: The name of the graph attribute which keeps the target module.
+    """
+    with graph.inserting_after(target_node):
+        return graph.create_node(
+            "call_module", module_attr_name, (target_node,), {}, name=module_attr_name + "_graph_node"
+        )
+
+
+def _get_target_node(graph: torch.fx.Graph, target_point: PTTargetPoint) -> torch.fx.Node:
+    """
+    Returns TorchFX graph node correspondent to the target point.
+
+    :param graph: Target torch.fx.Graph.
+    :param target_point: A target point to find the target node.
+    :return: TorchFX graph node correspondent to the target point.
+    """
+    target_type = target_point.target_type
+    target_node = FXModelTransformer.get_graph_node_by_name(graph, target_point.target_node_name)
+    if target_type in [TargetType.OPERATOR_PRE_HOOK, TargetType.OPERATION_WITH_WEIGHTS]:
+        target_node = target_node.all_input_nodes[target_point.input_port_id]
```

Comment:
Here I'm reusing PTTargetPoint class which supports only `input_port_id` parameters. For the post hooks it is working in the following way: new node is fed directly by the target node output and the hook output is used as the original node output.
So in your example one need to use target node B and input_port_id 0. This is the way to do it in PT backend so compression algorithms are adjusted to make OPERATOR_PRE_HOOK target points rather than OPERATOR_PRE_HOOK.

My guess is that post hooks port id were never used before thus never implemented for the PyTorch backend

Reply:
Let's discuss it offline.

Reply:
I've added a todo comment, I'll support such cases when the tests for model transformer will be introduced